### PR TITLE
Add Poetry configs for `bytes`, `boefjes`, `rocky`, and `mula`

### PR DESCRIPTION
**Pull request by Darwinkel**
### Changes
Implements Poetry dependency management for `bytes`, `boefjes`, `rocky`, and `mula`

Also removes an old requirements file in keiko and rocky, makes sure that the Poetry check workflow runs on changes in all relevant files and not just the project root, and slightly rewrites the Boefjes Dockerfile such that it doesn't intermix the hashed Poetry requirements and the unhashed module requirements. 

Unfortunately pip does not yet support hash checking mode when using git dependencies. This is a work in progress and should come later this year. Until then, we remove the hashes from rocky's requirements-dev.txt.

### Issue link
Closes https://github.com/minvws/nl-kat-coordination/issues/735

### Proof
See CI.

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: boefjes/Dockerfile [hunk 1]
```diff
@@ -29,6 +29,7 @@ RUN --mount=type=cache,target=/root/.cache \
     pip install --upgrade pip \
     && pip install -r /tmp/boefjes-requirements.txt \
     && rm /tmp/boefjes-requirements.txt \
+    && pip install -r requirements.txt \
```
Comment (Donnype) [inline boefjes/Dockerfile:32]: Could also be an `else` the line below although I don't expect much performance difference

**Review by Donnype — APPROVED** [review]
This looks good, was also wondering if we should/could use the poetry version to generate the right version for the APIs (where we do a sed now I believe). Perhaps not though ;)

**Comment by Darwinkel** [comment]
> This looks good, was also wondering if we should/could use the poetry version to generate the right version for the APIs (where we do a sed now I believe). Perhaps not though ;)

I looked into this, but it's non-trivial. https://pypi.org/project/poetry-dynamic-versioning/

I guess we could make a low-priority follow-up ticket to this.

File: boefjes/Dockerfile (hunk 1 above)
Comment (Darwinkel) [reply boefjes/Dockerfile:32]: We could, but it's done this way in all of our Dockerfiles and looks readable this way.

**Review by ammar92 — APPROVED** [review]
Looks good. Most stuff already discussed offline

**Comment by Donnype** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [ ] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [ ] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
Install still works fine, as expected given the changes.

### What doesn't work:
* _Likely unrelated_: I had some issues with the DNSReport where no Findings are in the list, but Findings are actually defined in the overview, but this could just be me having little experience creating DNS Reports:
image
image

* _Definitely unrelated_: I found a new bug with RabbitMQ in bytes that I will fix separately: we do not catch the issue of channels being closed

### Bug or feature?:
*

**Review by underdarknl — APPROVED** [review]

**Pull request merged**